modules/roles/client/services/roles.service.ts

```diff
@@ -27,17 +27,23 @@ export class RoleService {
   {
     return this.http
       .post('api/roles', formData)
-      .map(this.extractData);		
+      .map(this.formatCreateResponse);
   }
 
   removeRole(id: string): Observable<Role> {
     return this.http.delete('api/roles/' + id)
       .map((r: Response) => r.json().data);
   }
 
-  private extractData(res: Response | any) {
-    console.log(res);
+
+
+  private formatCreateResponse(res: Response | any) {
     let body = res.json();
+    if(body.error && body.message.indexOf("dup key") > 0)
```

Comment:
This seems a bit excessive of message handling.  Http errors should bubble up to a catch ( http://reactivex.io/rxjs/class/es6/Observable.js~Observable.html#instance-method-catch )

Which means you just have to implement this logic in the API and through the correct error code.  This would be preferable, because if we have to make another application (like a mobile app), this logic would have to be duplicated there.